felix/deltatracker/delta_tracker.go

```diff
@@ -43,6 +43,16 @@ import (
 // keys that are in the dataplane map but not in the desired map.  The
 // pending maps are exposed via the IterPendingUpdates and IterPendingDeletions
 // methods.
+//
+// It's not recommended to use mutable objects in a DeltaTracker.  Updating a
+// mutable object while it is in the DeltaTracker will corrupt its internal
+// state.  By design, the DeltaTracker aliases the Desired and Dataplane values
+// if they happen to be equal. So, to safely mutate a value, you must
+// take a copy, mutate the copy and then re-insert the value.  Note: it is not
+// safe to delete the object from one side of the tracker, mutate it and
+// re-insert it.  Deleting an object that is in-sync from the Dataplane side
```

Comment:
I don't follow this.  Does DeltaTracker provide an API for "delete the object from one side of the tracker"?  If it does, surely that ought to be coded safely?